### DISCUSSION TITLE
Exception handling with Dataframe of values that occur only once or twice

### DISCUSSION THREAD

**Discussion by romanwolf-git** [nmalkin/plot-likert#37] [General]
Hey, I am trying to plot a Dataframe in which values of the scale occur only once or twice. This always seems to run into a "ValueError: attempt to get argmax of an empty sequence". I can solve this issue by setting xtick_interval = 1. But still I don't understand the error, which would help in order to handle this error, since I am plotting more than 50 plots.

This is my DataFrame

<html><body>

Pkw | Fahrrad | ÖPNV | zu Fuß
-- | -- | -- | --
NaN | NaN | NaN | NaN
NaN | NaN | NaN | NaN
NaN | NaN | NaN | NaN
NaN | NaN | NaN | NaN
NaN | NaN | NaN | NaN
NaN | NaN | NaN | NaN
NaN | NaN | NaN | NaN
etwas mehr | etwas mehr | weniger | NaN
NaN | NaN | NaN | NaN
NaN | NaN | NaN | NaN
NaN | NaN | NaN | NaN
NaN | NaN | NaN | NaN
NaN | NaN | NaN | NaN
NaN | NaN | NaN | NaN
NaN | NaN | NaN | NaN
NaN | NaN | NaN | NaN
NaN | NaN | NaN | NaN
NaN | NaN | NaN | NaN
NaN | etwas mehr | etwas mehr | etwas mehr
NaN | NaN | NaN | NaN
NaN | NaN | NaN | NaN
NaN | NaN | NaN | NaN
NaN | NaN | NaN | NaN
NaN | NaN | NaN | NaN
NaN | NaN | NaN | NaN
NaN | NaN | NaN | NaN
NaN | NaN | NaN | NaN
NaN | NaN | NaN | NaN

</body>
</html>

If you could suggest any kind of help or a better explanantion, I would be very thankful!

**Comment by romanwolf-git** [comment]
By the way, this is how a Plot looks finally. Somehow the legend is shifted into the plotting canvas. If someone has a solution for that as well?!

![question_B222](https://user-images.githubusercontent.com/61413171/194267718-6179c524-aaeb-42ce-ab80-b86c0c1dbd81.png)

**Comment by alaskamike** [comment]
This line in plot_likert.py (in def plot_counts) should keep it out of the
plot:

    # Control legend
    plt.legend(bbox_to_anchor=(1.05, 1))

It seems like your plot has positive responses on the left and negative on
the right.  Maybe you need to change the anchor to -1.05?

Tschuess,
Mike

**Comment by nmalkin** [comment]
Hi @hbxbgrw7913, these are both bugs, thanks for reporting them! I've moved them to separate issues — #38 (for the original exception) and #39 (for the legend positioning) — and will follow up there.
